src/cpu/x64/jit_brgemm_inner_product_utils.cpp

```diff
@@ -525,7 +525,7 @@ status_t jit_brgemm_ip_fwd_conf_t::init_conf(cpu_isa_t isa,
     jbgp.nb_ic = div_up(jbgp.ic, jbgp.ic_block);
 
     // gemm-based inner product performs better when oc = 1
-    if (is_f32_compute && jbgp.oc == 1) return status::unimplemented;
+    if (is_f32_compute && jbgp.oc == 1 && jbgp.orig_wei_dt != f16) return status::unimplemented;
```

Comment:
I bet we should fix the condition not only for fp16 weights, but for all weights compression cases (jbgp.weights_decompression == true)

Reply:
Hi @dmitry-gorokhov Do you have any idea about how to create a test case with weights_decompression == true? Seems currently the weight decompression feature is not supported for OC==1 case ((https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/fullyconnected.cpp#L143))

Reply:
You are right, modelType is not configurable from the tests side.
Ok, lets just change `jbgp.orig_wei_dt != f16` on `!jbgp.weights_decompression`, check that the issue is fixed and merge w/o tests.

Reply:
Done. Thank you! @dmitry-gorokhov